src/reducers/wallet.js

```diff
@@ -1,13 +1,28 @@
-import { WALLET_READ_SUCCESS, WALLET_CREATE_SUCCESS } from '../actions';
+import { WALLET_READ_REQUEST, WALLET_READ_SUCCESS, WALLET_READ_FAILURE, WALLET_CREATE_SUCCESS } from '../actions';
 
-const initialState = [];
+const initialState = { list: [] };
 
 const reducer = (state = initialState, { type, payload }) => {
   switch (type) {
+    case WALLET_READ_REQUEST:
+      return { ...state, isLoading: true };
     case WALLET_READ_SUCCESS:
-      return [...state, ...payload];
+      return {
+        ...state,
+        list: payload,
+        isLoading: false,
+      };
```

Comment:
We received the list of wallets from the server as `payload`. We are simply swapping the list out with the newly received values, and resetting the `isLoading` flag to false.

I preferred not to include `isLoading: false` in the `initialState` cause both `undefined` and `false` are falsey values, but it provides us the flexibility to decide if there has ever been an API call made - if we'll need such info in the future.